patches/@rrweb__record@2.0.0-alpha.17.patch

```diff
@@ -1,8 +1,47 @@
 diff --git a/dist/record.js b/dist/record.js
-index 46ec389fefb698243008b39db65470dbdf0a3857..39d9ac821fd84f5a1494ac1e00c8d4e0e58c432a 100644
+index 46ec389fefb698243008b39db65470dbdf0a3857..7741b3c2c43cc868215dbd0ff1785557b158464a 100644
 --- a/dist/record.js
 +++ b/dist/record.js
-@@ -246,6 +246,9 @@ function isCSSImportRule(rule2) {
+@@ -20,6 +20,15 @@ const testableAccessors$1 = {
+   Element: ["shadowRoot", "querySelector", "querySelectorAll"],
+   MutationObserver: []
+ };
++const isFunction = (x) => typeof x === 'function';
++const isNativeFunction = () => isFunction(x) && x.toString().includes("[native code]");
++const isAngularZonePatchedFunction = (x) => {
++  if (!isFunction(x)) {
++      return false;
++  }
++  const prototypeKeys = Object.getOwnPropertyNames(x.prototype || {});
++  return prototypeKeys.some((key) => key.indexOf('__zone'));
++}
+ const testableMethods$1 = {
+   Node: ["contains", "getRootNode"],
+   ShadowRoot: ["getSelection"],
+@@ -35,14 +44,14 @@ function getUntaintedPrototype$1(key) {
+   const accessorNames = key in testableAccessors$1 ? testableAccessors$1[key] : void 0;
+   const isUntaintedAccessors = Boolean(
+     accessorNames && // @ts-expect-error 2345
+-    accessorNames.every(
+-      (accessor) => {
+-        var _a2, _b;
+-        return Boolean(
+-          (_b = (_a2 = Object.getOwnPropertyDescriptor(defaultPrototype, accessor)) == null ? void 0 : _a2.get) == null ? void 0 : _b.toString().includes("[native code]")
+-        );
+-      }
+-    )
++    accessorNames.every((accessor) => {
++      const candidate = Object.getOwnPropertyDescriptor(defaultPrototype, accessor)?.get;
++      let isUntainted = isNativeFunction(candidate);
++      if (key === 'MutationObserver') {
++        isUntainted = isUntainted && !isAngularZonePatchedFunction(candidate);
++      }
++      return isUntainted;
++    }),
```

Comment:
when testing mutation observer it has to be native code _and_ not have angular zone prototypes